### DISCUSSION TITLE
Relative date condition improvements

### DISCUSSION THREAD

**Discussion by rynpsc** [craftcms/cms#12003] [Ideas] [answered]
In 4.3 it will be possible to use relative dates within `BaseDateRangeConditionRule` which is a fantastic addition, however it only seems to support past dates or at least it's a bit unintuitive as you have to use a negative number (I think). In combination with "before" and "after" it comes a bit hard to reason with.

Would replacing the "Before" and "After" options with a single "Relative" option be more intuitive and then having the filter `inputHtml` setup with a dropdown with "Last/Past" and "Next" options followed by the number input and then the period dropdown?

Or keep "Before" and "After" but have them as a single date field and then have "Relative" as an addition.

I think the language of "Last/Past" and "Next" is easier to parse. For example if you wanted to filter by entries updated the in the last 3 days it's more natural to thing of that in terms of Date Updated -> Last -> 3 -> Days rather than Date Updated -> After -> 3 -> "days ago"

The last/next language is used in other programs with relative date filters.

- https://help.tableau.com/current/pro/desktop/en-us/qs_relative_dates.htm
- https://learn.microsoft.com/en-us/power-bi/visuals/desktop-slicer-filter-date-range

**Comment by brandonkelly** [comment, marked as answer]
> In 4.3 it will be possible to use relative dates within `BaseDateRangeConditionRule` which is a fantastic addition, however it only seems to support past dates or at least it's a bit unintuitive as you have to use a negative number (I think). In combination with "before" and "after" it comes a bit hard to reason with.

Agree that it should be easier to work with future dates. Just added new “minutes from now”, “hours from now”, and “days from now” period type options to the “Before…” and “After…” date range types: (62f92a4d9eae709ab37adc9a1429049e2df83dd9)

<img width="829" alt="A Post Date condition rule, set to “After…” → 30 → “minutes from now”" src="https://user-images.githubusercontent.com/47792/194066853-c35a69c2-0a39-436c-ba60-08737e762087.png">


> I think the language of "Last/Past" and "Next" is easier to parse. For example if you wanted to filter by entries updated the in the last 3 days it's more natural to thing of that in terms of Date Updated -> Last -> 3 -> Days rather than Date Updated -> After -> 3 -> "days ago"

“Last 3 days” would be a range from 3 days ago to the present, whereas “After 3 days ago” is any point in time after 3 days ago, including future dates. So those would solve different needs.

We can consider adding “In the last…” and “In the next…” range types down the road if there’s a definite need for them, but for now I’d rather not introduce the additional complexity.
